### PR TITLE
feat(serve): add CSS custom state toggle knobs

### LINKED ISSUE #410: serve: add CSS custom state toggle knobs
## Problem

The knobs panel in `cem serve` renders interactive controls for attributes, properties, and CSS custom properties, but does not render toggle controls for CSS custom states (`cssStates` in the manifest).

Elements with `@cssstate` declarations (e.g. `--loading`, `--invalid`) have no way to be toggled interactively in the dev server.

## Expected behavior

CSS custom states should appear as boolean toggle knobs in the knobs panel, allowing developers to add/remove states from the element's `ElementInternals.states` set.

## Implementation notes

- Add `KnobCategoryCSSState` to `knobs.go`
- Add `CSSStateKnobs []KnobData` field on `KnobsData`
- Generate boolean toggle knobs from `declaration.CssStates()`
- Client-side: toggle via `element.internals?.states.add()/delete()` or a fallback mechanism
- Wire into `generateKnobsForInstance` and `RenderKnobsHTML`
- Update docs as necessary.

Discovered while adding `@cssstate` to `examples/kitchen-sink`. Make sure at least one kitchen sink element implements `CustomStateSet`.

### PR DESCRIPTION
**Pull request by bennypowers**
## Summary

- Add interactive toggle controls for CSS custom states (`cssStates`) in the knobs panel
- Fix two serve routing bugs that caused all demo routes to 404 when using `serve -p`

## CSS State Knobs

- New `KnobCategoryCSSState` with boolean switch controls (`data-knob-type="css-state"`)
- Full event chain: `KnobCssStateChangeEvent` -> chrome coordination -> demo -> `CustomStateSet`
- Uses `globalThis._elementInternals` WeakMap (Web Components CG protocol, same as Deque Axe DevTools) for external access to `ElementInternals`
- Disables switch when element has no registered internals
- Knob order: Attributes > CSS Custom States > Properties > CSS Custom Properties

## Kitchen-sink demo-button refactor

- Implements `CustomStateSet` with `loading`, `active`, and `disabled` states
- Refactored CSS from shadow DOM classes to `:host(:state())` and `:host([attr])` selectors
- Spinner always in DOM, shown/hidden via `:host(:state(loading))` CSS
- Registers in `_elementInternals` WeakMap in constructor

## Serve routing fix (included as base commit)

- `BuildDemoRoutingTable` now skips invalid demo paths instead of failing the entire routing table
- Added `DemoURLPrefixFromTemplate()` to strip `urlTemplate` deployment prefix from local routes
- Both fixes threaded through `DevServerContext`

## Test plan

- [x] `TestCSSStateToKnob` -- boolean type, category
- [x] `TestGenerateKnobs_CSSStates` -- generation from manifest fixture
- [x] `TestRenderKnobsHTML_CSSStates` -- template renders switch controls
- [x] `TestParseEnabledKnobs` -- recognizes `css-states`
- [x] `TestDemoURLPrefixFromTemplate` -- prefix extraction edge cases
- [x] `TestExtractLocalRoute_PrefixStripping` -- deployment prefix stripped
- [x] `TestBuildDemoRoutingTable_DirectoryTraversalPrevention` -- bad entries skipped
- [x] All existing tests pass, golden files updated
- [x] Visual verification: knobs render in browser, toggles work

Closes #410

## Summary by CodeRabbit

* **New Features**
  * Added **CSS Custom States** (`css-states`) as new demo knobs with toggle UI, descriptions, and state application.
  * Updated demo knob wiring to support applying/clearing CSS custom states.
* **Bug Fixes**
  * Improved local demo routing/navigation under a URL prefix; invalid/unsupported demo entries are now skipped.
  * CSS state controls are disabled/unavailable when a demo can’t apply states.
* **Tests**
  * Expanded routing/navigation and knob test coverage, including URL-prefix extraction and CSS-state rendering.

### CODE REVIEW

File: examples/kitchen-sink/elements/demo-button/demo-button.css [hunk 1]
```diff
@@ -62,96 +62,113 @@
...
-  /* Appearances */
-  &.filled {
-    background: var(--demo-button-bg);
-    color: var(--demo-button-color);
-  }
+:host([appearance="plain"]) #button {
+  background: transparent;
+  color: var(--demo-button-bg);
+  border: none;
+}
```
[the remark this answers is not in the document]
Comment (bennypowers) [reply examples/kitchen-sink/elements/demo-button/demo-button.css:119]: Fixed in 8ec70add -- appearance rules now nest `#button` inside each `:host([appearance="..."])` block.

File: examples/kitchen-sink/elements/demo-button/demo-button.css [hunk 2]
```diff
@@ -62,96 +62,113 @@
...
+
+:host(:state(loading)) #button {
+  pointer-events: none;
+}
+
+:host(:state(active)) #button {
+  transform: scale(0.96);
+  box-shadow: inset 0 2px 4px rgb(0 0 0 / 25%);
+}
+
```
[the remark this answers is not in the document]
Comment (bennypowers) [reply examples/kitchen-sink/elements/demo-button/demo-button.css:178]: Fixed in 8ec70add -- state rules now nest descendants inside their `:host` blocks. Removed dead `@cssstate active` (nothing wires it). Filed #417 for the cssProperties defaults issue (pre-existing, not from this PR).

File: serve/elements/cem-serve-chrome/cem-serve-chrome.ts [hunk 3]
```diff
@@ -1427,6 +1432,22 @@ Generated: ${new Date().toISOString()}`;
...
+        `[data-knob-type="css-state"][data-knob-name="${CSS.escape(name)}"]`,
+      ) as HTMLInputElement | null;
+      if (control) {
+        control.checked = false;
+        control.disabled = true;
+      }
+      break;
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (bennypowers) [reply serve/elements/cem-serve-chrome/cem-serve-chrome.ts:1457]: Fixed in 8ec70add -- `#disableCssStateKnob` now hides the clear button and adds a descriptive title. Also fixed the false-disable issue: only disables when the element exists but lacks internals, not when the target instance is missing.

**Comment by bennypowers** [comment]
@coderabbitai review
